tools/lib/filereader.py

```diff
@@ -1,13 +1,29 @@
 import os
+import socket
+from urllib.parse import urlparse
 from openpilot.tools.lib.url_file import URLFile
 
 DATA_ENDPOINT = os.getenv("DATA_ENDPOINT", "http://data-raw.comma.internal/")
 
+
+def is_data_endpoint_accessible():
+  hostname = urlparse(DATA_ENDPOINT).hostname
+  if not hostname:
+    return False
+
+  try:
+    _ = socket.gethostbyname(hostname)
+    return True
+  except OSError:
+    return False
+
```

Comment:
this can return None

Reply:
good catch

Reply:
hmm, if it throws anything other than OSError, this will be rethrown to the caller, which is expected